V0.1.1 : pytest for BigQueryToCSV.extract()

tusharchou opened the pull request:
# Please review this test and the code changes along with it

<img width="1014" alt="Screenshot 2024-11-05 at 12 44 49 PM" src="https://github.com/user-attachments/assets/e02cf83d-cdd4-4428-be22-19a1a9b10560">


---
name: Pull request
about: Implement an solution for this project
title: ''
labels: ''
assignees: ''

---

**Is your pull request related to a issue? Please tag.**

**Describe the solution you have implemented**

**Describe why alternatives you considered are not optimal**
A clear and concise description of any alternative solutions or features you've considered.

**Additional context**
Add any other context or screenshots about the pull request here.

tusharchou committed:
fixed bug warehouse uri

Thu Oct 24 9:25 PM IST

tusharchou committed:
added logging to CSV.get()

tusharchou committed:
supported big query ts format

tusharchou committed:
refactored parameter to config from catalog to reduce confusion

tusharchou committed:
Fixed bug of logger in GCP

tusharchou committed:
replaced local path with dynamic path

tusharchou committed:
replaced local path with dynamic path

tusharchou committed:
replaced local path with dynamic path

tusharchou committed:
demo

tusharchou committed:
demo

tusharchou committed:
added a class implementation for github issue hoping it is useful for guiding users to resolution ETAs eventually

tusharchou committed:
added a exception for PlanNotFound to ask users to raise issues on the repository for resolution

tusharchou committed:
Updated overview and milestones. Added directory structure under technical specifications.

tusharchou committed:
Updated components in technical speciifcations

tusharchou committed:
added testing for BigQueryToCSV.extract()

tusharchou committed:
added testing for BigQueryToCSV.extract()

tusharchou committed:
added testing for Iceberg.get()

mrutunjay-kinagi commented:
Please Add Test case for the following functionalities : 
1. BigQuery.get()
2. BigQuery.put()
3. CSV.get()
4. CSV.put()
5. Iceberg.get()
6. Iceberg.put()
7. Parquet.get()
8. Parquet.put()

tusharchou committed:
added testing for table_statistics_from_iceberg_metadata

tusharchou commented on local-data-platform/tests/test_metadata.py:
```diff
@@ -0,0 +1,75 @@
+        session_catalog=catalog,
+        identifier = identifier
+    )
+    stats = table_statistics_from_iceberg_metadata(
```
#1 I am trying to solve for the statistics here @rakhioza07 @redpheonixx

tusharchou committed:
added table_statistics_from_iceberg_metadata in  format.iceberg.metadata

tusharchou committed:
added vocab in product

tusharchou committed:
added dimension in store

tusharchou committed:
tested dimension class without file format

tusharchou commented on local-data-platform/tests/test_dimension.py:
```diff
@@ -0,0 +1,23 @@
+from local_data_platform.store.dimension import Dimension
+from local_data_platform.product.vocab import Vocab
+
+def test_get():
```
Love how simple this is 🥇

tusharchou committed:
work: doc
topic: pyiceberg
subject: table.metadata.partition.count()
to-do: pytest table.count()
eta: Nov 28

tusharchou committed:
work: testing
topic: pyiceberg
subject: table.scan().count()
to-do: implement table.scan().count()
eta: Nov 28

tusharchou committed:
kafka spark iceberg WIP

tusharchou commented:
I will work on this
